### ISSUE TITLE
Running putting object twice gives: Aws::Errors::NoSuchEndpointError: Encountered a `SocketError` while attempting to connect to:

### ISSUE THREAD

**Issue by lephyrius** [aws/aws-sdk-ruby#1492]
When I run Amazon S3 client.  I can only make 1 request and then after that it creates an error:
```Aws::Errors::NoSuchEndpointError: Encountered a `SocketError` while attempting to connect to:```
```
/usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.9.11/lib/seahorse/client/plugins/raise_response_errors.rb:15:in `call'
/usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.9.11/lib/aws-sdk-core/plugins/s3_sse_cpk.rb:19:in `call'
/usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.9.11/lib/aws-sdk-core/plugins/s3_dualstack.rb:24:in `call'
/usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.9.11/lib/aws-sdk-core/plugins/s3_accelerate.rb:34:in `call'
/usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.9.11/lib/aws-sdk-core/plugins/jsonvalue_converter.rb:20:in `call'
/usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.9.11/lib/aws-sdk-core/plugins/idempotency_token.rb:18:in `call'
/usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.9.11/lib/aws-sdk-core/plugins/param_converter.rb:20:in `call'
/usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.9.11/lib/seahorse/client/plugins/response_target.rb:21:in `call'
/usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.9.11/lib/seahorse/client/request.rb:70:in `send_request'
/usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.9.11/lib/seahorse/client/base.rb:207:in `block (2 levels) in define_operation_methods'
```
Here is the code that is causing it:
```ruby
  def s3
    @s3 ||= Aws::S3::Client.new(region: 'us-east-1', access_key_id: 'id', secret_access_key: 'secret'
  end
output = 'awesome data'
s3.put_object(bucket: 'data-2000',key: filename ,body: output)
```
I always have the same client why does it think I have different regions? Do I have to recreate the client with each request?

**Comment by joshuasiler** [comment]
It looks like you are running ruby 2.4, which has known issues with this gem. Not sure if that's the problem but it seems like a candidate. 

Some folks are working on it https://github.com/aws/aws-sdk-ruby/issues/1461

**Comment by awood45** [comment]
I definitely don't see anything in that usage example that should be an issue. Going to work on repro with Ruby 2.4.

**Comment by cjyclaire** [comment]
Adding notes, this failure is caused by [`webmock`](https://github.com/bblimke/webmock/blob/master/CHANGELOG.md#231), their support for ruby 2.4 landed in webmock 2.3.1. If you try `bundle update`, should help fix the problem.

Meanwhile, other 2.4 refractor work is tracked at #1505, we are working on getting it merged soon : )

**Comment by cjyclaire** [comment]
PR #1505 has been merged and will be tagged in the next release, feel free to reopen if you still have problem with it : )
